src/coral_models/prepare_raw_data.py

```diff
@@ -234,19 +237,41 @@ def make_recording_metadata(
         # no information about the recorders for the read aloud data.
         read_aloud_data["recorder_id"] = -1
 
-        # Drop columns that are not in the recording metadata
-        read_aloud_data = read_aloud_data[recording_metadata.columns]
+        read_aloud_data["speaker_id"] = "placeholder"
 
         # Append to recording metadata list
         recording_metadata_list.append(read_aloud_data)
 
-    # Concatenate all recording metadata
-    all_recording_metadata = (
+    # Concatenate all read aloud recording metadata
+    all_read_recording_metadata = (
         pd.concat(recording_metadata_list, axis=0)
         .drop_duplicates()
         .reset_index(drop=True)
     )
 
+    # People often typed their name in all lower case or similar variations, which
+    # yields a lot of duplicates. We therefore remove rows with duplicate emails.
+    all_read_recording_metadata["name"] = all_read_recording_metadata["email"].map(
+        mail_to_name
+    )
+
+    # Convert speaker emails to speaker IDs
+    all_read_recording_metadata["speaker_id"] = all_read_recording_metadata[
+        ["name", "email"]
+    ].apply(lambda x: speaker_id(x[0], x[1]), axis=1)
+
+    # Drop columns that are not in the recording metadata
+    all_read_recording_metadata = all_read_recording_metadata[
+        recording_metadata.columns
+    ]
+
+    # Concatenate the read aloud recording metadata with the other recording metadata
+    all_recording_metadata = (
+        pd.concat([all_read_recording_metadata, recording_metadata], axis=0)
+        .drop_duplicates()
+        .reset_index(drop=True)
+    )
+
     # Make a recording id column
     tqdm.pandas()
     all_recording_metadata["recording_id"] = all_recording_metadata[
```

Comment:
This could possibly be a separate `postprocess_recording_metadata` function. Again to split things up.

Reply:
Refactoring in another PR.